src/tt-gdb.py

```diff
@@ -0,0 +1,401 @@
+        ( 'space::child_fk_constraint', 'fk_constraint::link' ),
+        ( 'space::ck_constraint', 'ck_constraint::link' ),
+        ( 'space::memtx_stories', 'memtx_story::in_space_stories' ),
+        ( 'space::on_replace', 'trigger::link' ),
+        ( 'space::parent_fk_constraint', 'fk_constraint::link' ),
+        ( 'space::space_cache_pin_list', 'space_cache_holder::link' ),
+        ( 'tx_manager::read_view_txs', 'txn::in_read_view_txs' ),
+        ( 'tx_manager::all_stories', 'memtx_story::in_all_stories' ),
+        ( 'tx_manager::traverse_all_stories', 'memtx_story::in_all_stories' ), # struct rlist*
+        ( 'tx_manager::all_txs', 'txn::in_all_txs' ),
+        ( 'txn::conflict_list', 'tx_conflict_tracker::in_conflict_list' ),
+        ( 'txn::conflicted_by_list', 'tx_conflict_tracker::in_conflicted_by_list' ),
+        ( 'txn::gap_list', 'gap_item::in_gap_list' ),
+        ( 'txn::full_scan_list', 'full_scan_item::in_full_scan_list' ),
+        ( 'txn::on_commit', 'trigger::link' ),
+        ( 'txn::on_rollback', 'trigger::link' ),
+        ( 'txn::on_wal_write', 'trigger::link' ),
+        ( 'txn::point_holes_list', 'point_hole_item::in_point_holes_list' ),
+        ( 'txn::read_set', 'tx_read_tracker::in_read_set' ),
+        ( 'txn::savepoints', 'txn_savepoint::link' ),
+        ( 'txn_limbo::queue', 'txn_limbo_entry::in_queue' ),
+        ( 'txn_stmt::on_commit', 'trigger::link' ),
+        ( 'txn_stmt::on_rollback', 'trigger::link' ),
+        ( 'sql_stmt_cache::gc_queue', 'stmt_cache_entry::link' ),
+        ( 'sys_alloc::allocations', 'container::rlist' ),
+        ( 'user::credentials_list', 'credentials::in_user' ),
+        ( 'vy_cache_env::cache_lru', 'vy_cache_node::in_lru' ),
+        ( 'vy_history::stmts', 'vy_history_node::link' ),
+        ( 'vy_join_ctx::entries', 'vy_join_entry::in_ctx' ),
+        ( 'vy_lsm::on_destroy', 'trigger::link' ),
+        ( 'vy_lsm::runs', 'vy_run::in_lsm' ),
+        ( 'vy_lsm::sealed', 'vy_mem::in_sealed' ),
+        ( 'vy_lsm_recovery_info::ranges', 'vy_range_recovery_info::in_lsm' ),
+        ( 'vy_lsm_recovery_info::runs', 'vy_run_recovery_info::in_lsm' ),
+        ( 'vy_quota::wait_queue', 'vy_quota_wait_node::in_wait_queue' ),
+        ( 'vy_range::slices', 'vy_slice::in_range' ),
+        ( 'vy_range_recovery_info::slices', 'vy_slice_recovery_info::in_range' ),
+        ( 'vy_recovery::lsms', 'vy_lsm_recovery_info::in_recovery' ),
+        ( 'vy_tx::on_destroy', 'trigger::link' ),
+        ( 'vy_tx_manager::read_views', 'vy_read_view::in_read_views' ),
+        ( 'vy_tx_manager::writers', 'vy_tx::in_writers' ),
+        ( 'vy_write_iterator::src_list', 'vy_write_src::in_src_list' ),
+        ( 'wal_writer::watchers', 'wal_watcher::next' ),
+        ( 'watchable::pending_watchers', 'watcher::in_idle_or_pending' ),
+        ( 'watchable_node::all_watchers', 'watcher::in_all_watchers' ),
+        ( 'watchable_node::idle_watchers', 'watcher::in_idle_or_pending' ),
```

Comment:
I doubt we'll manage to keep this database in sync. AFAICS, in [Linux kernel scripts](https://github.com/torvalds/linux/blob/master/scripts/gdb/linux/lists.py), they don't maintain a database, but instead require the caller to pass the link name to the iterator explicitly, like one does in C code. Maybe, we could do the same?

Reply:
The only reason this database was introduced is to simplify working with rlists. From the point in gdb where one want to display some rlist it's not always trivial to get actual entry type and field it's linked by. I agree that keeping database in sync will require some additional attention, but it doesn't seem to be an operation that must be done on _every_ change that affects rlist. How about introducing additional arguments that specify explicitly entry info (struct and field), but keep database to try to recognize them automatically if omitted? We can then check and sync database not too often... only when automatic recognition is missing too often

Reply:
Added optional argument to specify entry information explicitly

Reply:
Personally I like that, say, `tt-list cord_ptr->alive` just works.